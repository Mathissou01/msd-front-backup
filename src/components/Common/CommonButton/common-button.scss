@use "src/styles/01-settings/settings.colors" as colors;
@use "src/styles/02-tools/tools.include-media" as im;
@use "src/styles/02-tools/tools.buttons" as buttons;
@use "src/styles/02-tools/tools.typography" as typo;
@use "src/styles/02-tools/tools.functions" as func;

.c-CommonButton {
  &_fontSmall {
    @include typo.text("buttonSmall");
  }

  &_fontLarge {
    @include typo.text("buttonLarge");
  }

  &_primary {
    @include buttons.button("primary");
  }

  &_primaryContrast {
    @include buttons.button("primaryContrast");
  }

  &_secondary {
    @include buttons.button("secondary");
  }

  &_tertiary {
    @include buttons.button("tertiary");
  }

  &_lightGray {
    @include buttons.button("lightGray");
  }

  &_paddingSmall {
    padding: 8px 16px;
  }

  &_paddingLarge {
    padding: 15px;
  }

  &_fullWidth {
    @include im.media("<laptop") {
      width: 100%;
    }
  }

  &_isRounded {
    border-radius: 50%;
  }

  &_disabled {
    outline: none !important;
    cursor: not-allowed;
  }

  &__Picto {
    width: 24px;
    height: 24px;

    &_search {
      background-image: url("data:image/svg+xml,%3Csvg width='24' height='24' viewBox='0 0 24 24' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cg clip-path='url(%23a)' stroke='#{func.hex-to-url-string(colors.$contrast-text)}' stroke-width='1.5' stroke-linecap='round' stroke-linejoin='round'%3E%3Cpath d='M1.473 13.357a9.063 9.063 0 1 0 16.682-7.09 9.063 9.063 0 0 0-16.682 7.09v0ZM16.22 16.22l7.03 7.03'/%3E%3C/g%3E%3Cdefs%3E%3CclipPath id='a'%3E%3Cpath fill='%23fff' d='M0 0h24v24H0z'/%3E%3C/clipPath%3E%3C/defs%3E%3C/svg%3E");
    }

    &_cross {
      background-image: url("data:image/svg+xml,%3Csvg width='24' height='24' viewBox='0 0 24 24' fill='#{func.hex-to-url-string(colors.$contrast-text)}' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='m.75 23.249 22.5-22.5m0 22.5L.75.749' stroke='#{func.hex-to-url-string(colors.$contrast-text)}' stroke-width='1.5' stroke-linecap='round' stroke-linejoin='round'/%3E%3C/svg%3E%0A");
    }

    &_target {
      background-image: url("data:image/svg+xml,%3Csvg width='24' height='24' viewBox='0 0 16 16' fill='#{func.hex-to-url-string(colors.$contrast-text)}' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath stroke-width='0.1' d='M8.00049 15.998C7.72449 15.998 7.50049 15.774 7.50049 15.498V13.9767C4.60715 13.7347 2.26382 11.3907 2.02249 8.49805H0.500488C0.224488 8.49805 0.000488281 8.27405 0.000488281 7.99805C0.000488281 7.72205 0.224488 7.49805 0.500488 7.49805H2.02249C2.26516 4.60538 4.60849 2.26271 7.50049 2.02071V0.498047C7.50049 0.222047 7.72449 -0.00195312 8.00049 -0.00195312C8.27649 -0.00195312 8.50049 0.222047 8.50049 0.498047V2.02071C11.3932 2.26271 13.7365 4.60605 13.9785 7.49805H15.5005C15.7765 7.49805 16.0005 7.72205 16.0005 7.99805C16.0005 8.27405 15.7765 8.49805 15.5005 8.49805H13.9792C13.7372 11.3914 11.3938 13.7347 8.50049 13.9767V15.498C8.50049 15.774 8.27649 15.998 8.00049 15.998ZM3.02582 8.49805C3.26182 10.838 5.16049 12.7367 7.50049 12.9734V11.498C7.50049 11.222 7.72449 10.998 8.00049 10.998C8.27649 10.998 8.50049 11.222 8.50049 11.498V12.9734C10.8405 12.7367 12.7385 10.8387 12.9752 8.49805H11.5005C11.2245 8.49805 11.0005 8.27405 11.0005 7.99805C11.0005 7.72205 11.2245 7.49805 11.5005 7.49805H12.9752C12.7385 5.15805 10.8405 3.26071 8.50049 3.02405V4.49805C8.50049 4.77405 8.27649 4.99805 8.00049 4.99805C7.72449 4.99805 7.50049 4.77405 7.50049 4.49805V3.02405C5.16049 3.26071 3.26249 5.15805 3.02582 7.49805H4.50049C4.77649 7.49805 5.00049 7.72205 5.00049 7.99805C5.00049 8.27405 4.77649 8.49805 4.50049 8.49805H3.02582Z' stroke='#{func.hex-to-url-string(colors.$contrast-text)}'/%3E%3C/svg%3E%0A");
    }

    &_download {
      background-image: url("data:image/svg+xml,%3Csvg width='24' height='24' viewBox='9 11 18 15' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cg clip-path='url(%23a)' stroke-width='1.5' stroke-linecap='round' stroke-linejoin='round'%3E%3Cpath d='M18 12.74v8' stroke='%23030F40'/%3E%3Cpath d='m15 17.74 3 3 3-3m4.5 3v1a2 2 0 0 1-2 2h-11a2 2 0 0 1-2-2v-1' stroke='%23030F40'/%3E%3C/g%3E%3Cdefs%3E%3CclipPath id='a'%3E%3Cpath fill='%23fff' d='M10 10.24h16v16H10z'/%3E%3C/clipPath%3E%3C/defs%3E%3C/svg%3E");
    }
  }
}
