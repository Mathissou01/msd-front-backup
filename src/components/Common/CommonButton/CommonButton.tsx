import React from "react";
import "./common-button.scss";

interface ICommonButtonProps {
  label?: string;
  type?: "button" | "submit" | "reset" | undefined;
  picto?: "search";
  isDisabled?: boolean;
  onClick?: () => void;
  formLabelId?: string;
  buttonRef?: React.RefObject<HTMLButtonElement>;
  style?: "primary" | "secondary" | "tertiary" | null;
  paddingStyle?: "paddingDefault" | "paddingSmall" | "paddingLarge";
  isFullWidth?: boolean;
}

export default function CommonButton({
  label,
  picto,
  type = "button",
  isDisabled = false,
  onClick,
  formLabelId,
  buttonRef,
  style = "secondary",
  paddingStyle = "paddingDefault",
  isFullWidth = false,
}: ICommonButtonProps) {
  return (
    <button
      className={`c-CommonButton ${style ? "c-CommonButton_" + style : ""} ${
        paddingStyle !== "paddingDefault"
          ? "c-CommonButton_" + paddingStyle
          : ""
      } ${isFullWidth ? "c-CommonButton_fullWidth" : ""}`}
      type={type}
      onClick={onClick}
      disabled={isDisabled}
      id={formLabelId}
      ref={buttonRef}
    >
      {picto && (
        <div
          className={`c-CommonButton__Picto c-CommonButton__Picto_${picto}`}
        />
      )}
      {label}
    </button>
  );
}
