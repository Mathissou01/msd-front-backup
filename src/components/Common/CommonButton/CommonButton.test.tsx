import { fireEvent, render, screen } from "@testing-library/react";
import CommonButton from "./CommonButton";

const mock = {
  label: "common button",
};

it("renders", () => {
  const { container } = render(
    <CommonButton label={mock.label} style={null} />,
  );
  expect(container).toMatchSnapshot();
});

it("has working props", () => {
  const handleClick = jest.fn();
  render(
    <CommonButton
      label={mock.label}
      style={"primary"}
      type={"button"}
      isDisabled={false}
      onClick={handleClick}
      formLabelId={"common-button"}
    />,
  );
  const button = screen.getByRole("button");
  expect(button).toBeInTheDocument();
  expect(button).toHaveTextContent("common button");
  expect(button).toHaveClass("c-CommonButton_primary");
  fireEvent.click(button);
  expect(handleClick).toHaveBeenCalledTimes(1);
});

it("can be disabled", () => {
  const handleClick = jest.fn();
  render(
    <CommonButton label={mock.label} isDisabled={true} onClick={handleClick} />,
  );
  const button = screen.getByRole("button");
  expect(button).toHaveAttribute("disabled");
  fireEvent.click(button);
  expect(handleClick).toHaveBeenCalledTimes(0);
});
