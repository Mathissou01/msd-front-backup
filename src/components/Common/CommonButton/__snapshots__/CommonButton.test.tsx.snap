// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <button
    class="c-CommonButton   "
    type="button"
  >
    common button
  </button>
</div>
`;
